### PR TITLE
added the category filter

### PR DESCRIPTION
**Pull request by anshikaa12**
add category filter!
ecomm_filterrcatReactt

### CODE REVIEW

File: src/services/category_api.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import {
+    useEffect,
+    useState
+} from "react";
+import axios from "axios";
+
+function useCategory() {
+    const [category, setCategory] = useState([]);
+    useEffect(() => axios.get('/api/categories').then(resp => setCategory(resp.data.categories).catch(err => console.error(err))), []);
```
Comment (Vishruta-Patil) [inline src/services/category_api.js]: Use a separate file for the api calls.  
Its good practice to move the API calls into the service folder and then get that function into a different file

**Review by Vishruta-Patil — APPROVED** [review]
Everything else looks good to merge

File: src/components/Card.jsx [hunk 2]
```diff
@@ -5,14 +5,19 @@ import { useWishlist } from "../context/wishlistcontext";
 import { useFilter } from '../context/filtercontext';
 import { sortedData } from "../context/filtercontext"
 import { sorted_price_list } from "../reducer/filterreducer";
+import { sorted_category_list } from "../reducer/filterreducer";
+import { useCategory } from '../services/category_api';
```
Comment (arp99) [inline src/components/Card.jsx]: Suggestion: Be consistent with your naming convention you are following

File: src/components/Card.jsx [hunk 3]
```diff
@@ -5,14 +5,19 @@ import { useWishlist } from "../context/wishlistcontext";
...
+
 
 function Card() {
     const { state, dispatch } = useCart();
     const { cardData } = useProduct();
+    const { category } = useCategory();
     const { currState, wishFunc } = useWishlist();
     const { filterState, filterFunc } = useFilter();
```
Comment (arp99) [inline src/components/Card.jsx]: Suggestion: Be consistent with naming states and dispatch
- In cart context change the name fo state to `cart` and dispatch to `cartFunc`, as you have done the same thing for wishlist and Filter

File: src/components/Card.jsx [hunk 4]
```diff
@@ -5,14 +5,19 @@ import { useWishlist } from "../context/wishlistcontext";
...
-    const sortedData = sorted_price_list(filterState.sortBy, cardData);
-    const productsList = filterState.sortBy !== "" ? sortedData : cardData;
+    const sortedData = filterState.sortBy !== "" ? sorted_price_list(filterState.sortBy, cardData) : cardData;
+    const sorted_category_data = sorted_category_list(filterState.categories, sortedData);
```
Comment (arp99) [inline src/components/Card.jsx]: Suggestion: Same here, maintain consistent naming convention

File: src/pages/product_listing/Filter.jsx [hunk 5]
```diff
@@ -33,30 +35,13 @@ function Filter() {
...
-                    <input type="checkbox" />
-                    <span className="checkmark"></span>
-                </label>
+                {category.map(item => {
+                    return <label className="check-list" key={item._id}>{item.categoryName}
+                        <input type="checkbox" checked={filterState.categories && filterState.categories.includes(item.categoryName)} onChange={() =>
+                            filterFunc({ type: "SORT_CATEGORY", payload: item.categoryName })} />
+                        <span className="checkmark"></span>
+                    </label>
+                })}
```
Comment (arp99) [inline src/pages/product_listing/Filter.jsx]: Suggestion: Use prettier to indent all your codes properly

File: src/services/category_api.js (hunk 1 above)
Comment (anshikaa12) [reply src/services/category_api.js]: i am actually using a different folder for that!

**Review by arp99 — APPROVED** [review]
Looks good

**Pull request merged**